### PR TITLE
enh(profile): Refactor code to use vue components and styles where possible

### PR DESCRIPTION
**Pull request by susnux**
## Summary
Refactor profile view to make use of standard vue components. This reduces code complexity and duplicated code.
Moreover some visual bugs are resolved, e.g. the profile action buttons now work correctly (see below).

Also a lot of custom styling is removed in favor of our component styles from the nextcloud vue library.

actions before | actions after
---|---
Screenshot_20240109_194640|Screenshot_20240109_203810

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by emoral435** [comment]
unrelated comment to the PR - wish I could review this PR while at University because it looks like a good change, but my computer has 3% 😔 will try to review later if not merged !

**Review by Pytal — APPROVED** [review]
🖖

**Review by emoral435 — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Comment by nickvergessen** [comment]
>  OpenAPI / openapi (pull_request) Failing after 1m


Merged with red CI breaking it.

File: core/Controller/ProfilePageController.php [hunk 1]
```diff
@@ -42,7 +42,7 @@
 use OCP\Share\IManager as IShareManager;
 use OCP\UserStatus\IManager as IUserStatusManager;
 
-#[IgnoreOpenAPI]
+#[OpenAPI(scope: OpenAPI::SCOPE_IGNORE)]
```
Comment (nickvergessen) [inline core/Controller/ProfilePageController.php:45]: Sadly this is not supported by the tool yet.... Still waiting for @provokateurin in https://github.com/nextcloud/openapi-extractor/pull/24

Comment (nickvergessen) [reply core/Controller/ProfilePageController.php:45]: Fix in https://github.com/nextcloud/server/pull/42705

**Pull request merged**